src/main/java/controller/AllCaregiverController.java

```diff
@@ -0,0 +1,167 @@
+
+        this.tableView.setItems(this.tableviewContent);
+    }
+
+    /**
+     * handles new firstname value
+     * @param event event including the value that a user entered into the cell
+     */
+    @FXML
+    public void handleOnEditFirstname(TableColumn.CellEditEvent<Caregiver, String> event) {
+        event.getRowValue().setFirstName(event.getNewValue());
+        doUpdate(event);
+    }
+
+    /**
+     * handles new surname value
+     * @param event event including the value that a user entered into the cell
+     */
+    @FXML
+    public void handleOnEditSurname(TableColumn.CellEditEvent<Caregiver, String> event) {
+        event.getRowValue().setSurname(event.getNewValue());
+        doUpdate(event);
+    }
+
+    /**
+     * handles new telephone value
+     * @param event event including the value that a user entered into the cell
+     */
+    @FXML
+    public void handleOnEditTelephone(TableColumn.CellEditEvent<Caregiver, String> event) {
+        event.getRowValue().setTelephone(event.getNewValue());
+        doUpdate(event);
+    }
+
+    /**
+     * updates a caregiver by calling the update-Method in the {@link CaregiverDAO}
+     * @param t row to be updated by the user (includes the caregiver)
+     */
+    private void doUpdate(TableColumn.CellEditEvent<Caregiver, String> t) {
+        try {
+            dao.update(t.getRowValue());
+        } catch (SQLException e) {
+            e.printStackTrace();
+        }
+    }
+
+    /**
+     * calls readAll in {@link CaregiverDAO} and shows caregiver in the table
+     */
+    private void readAllAndShowInTableView() {
+        this.tableviewContent.clear();
+        this.dao = DAOFactory.getDAOFactory().createCaregiverDAO();
+        List<Caregiver> allCaregiver;
+        try {
+            allCaregiver = dao.readAll();
+            for (Caregiver c: allCaregiver) {
+                this.tableviewContent.add(c);
+            }
+        } catch (SQLException e) {
+            e.printStackTrace();
+        }
+
+    }
+
+    /**
+     * handles a delete-click-event. Calls the delete methods in the {@link CaregiverDAO}}
+     */
+    @FXML
+    public void handleDeleteRow() {
+        Caregiver selectedItem = this.tableView.getSelectionModel().getSelectedItem();
+        try {
+            dao.deleteById(selectedItem.getCid());
+            this.tableView.getItems().remove(selectedItem);
+        } catch (SQLException e) {
+            e.printStackTrace();
+        }
+    }
+
+    /**
+     * handles a add-click-event. Creates a caregiver and calls the create method in the {@link CaregiverDAO}
+     */
+    @FXML
+    public void handleAdd() {
+        String surname = this.txfSurname.getText();
+        String firstname = this.txfFirstname.getText();
+        String telephone = this.txfTelephone.getText();
+        try {
+            Caregiver c = new Caregiver(firstname, surname, telephone);
+            dao.create(c);
```

Comment:
```suggestion
            dao.create(new Caregiver(firstname, surname, telephone));
```
Auch hier wird wieder unnötig Speicher reserviert